feat(core): add customizable `LoggerContext` with labeling support (#4233)
mikro-orm/mikro-orm: docs/docs/logging.md

});
```

### Using a custom `LoggerFactory`

If you want more control over what is logged and how, use the `loggerFactory` option in your config and either extend the `DefaultLogger` class or make your `Logger` from scratch:

#### Extending `DefaultLogger`
We can extend the `DefaultLogger` instead of implementing everything from scratch. It is also exported from the `@mikro-orm/core` package:

```ts
class CustomLogger extends DefaultLogger {
  log(namespace: LoggerNamespace, message: string, context?: LogContext) {
    // Create your own implementation for output:
    console.log(`[${namespace}] (${context.label}) ${message}`);

    // OR Utilize DefaultLogger's implementation:
    super.log(namespace, message, context)
  }
}

return MikroORM.init({
  debug: true,
  loggerFactory: (options) => new CustomLogger(options),
});
```

> #### Creating a custom logger from scratch
> We can use `loggerFactory` and use our own implementation of the `Logger` interface:

Reviewer: ```suggestion
#### Creating a custom logger from scratch

You can use `loggerFactory` and use your own implementation of the `Logger` interface:
```

});
```

> ### Using a custom `LoggerFactory`

Reviewer: keep a blank line around headings (all of them)

```suggestion
### Using a custom `LoggerFactory`

```

Author: Whoops, good catch... I had to disable my auto-formatting because what I use for other projects is _**very**_ different from yours (I don't even like mine but at least it's consistent with format-on-save haha)

If you want more control over what is logged and how, use the `loggerFactory` option in your config and either extend the `DefaultLogger` class or make your `Logger` from scratch:

#### Extending `DefaultLogger`
We can extend the `DefaultLogger` instead of implementing everything from scratch. It is also exported from the `@mikro-orm/core` package:

```ts
class CustomLogger extends DefaultLogger {
  log(namespace: LoggerNamespace, message: string, context?: LogContext) {
    // Create your own implementation for output:
    console.log(`[${namespace}] (${context.label}) ${message}`);

    // OR Utilize DefaultLogger's implementation:
    super.log(namespace, message, context)
  }
}

return MikroORM.init({
  debug: true,
  loggerFactory: (options) => new CustomLogger(options),
});
```

> 

Reviewer: should be just a single blank line here
